apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -186,6 +188,20 @@ fn setup_metrics_exporter<T: MetricsConfigurator>(
 
 impl Drop for Telemetry {
     fn drop(&mut self) {
+        // If we can downcast the metrics exporter to be a `BasicController`, then we
+        // should stop it to ensure metrics are transmitted before the exporter is dropped.
+        for exporter in self.metrics_exporters.drain(..) {
```

Comment:
something is unclear to me here. I thought the metrics were held under the `METRICS_LAYER_HANDLE` singleton, which would typically survive after Telemetry drops. What's the lifecycle here, who gets created and dropped when?

Reply:
METRICS_LAYER_HANDLE is only preserving prometheus metrics. It doesn't preserve OTLP (push style) metrics. 

This draft PR is an experiment to see if we can force push metrics when an exporter is dropped, hence the check to see if we can downcast to `BasicController`. If I can show this is working then I'll need to think more deeply about how best to integrate this into the code base.